subprojects/core-api/src/main/java/org/gradle/api/file/FileContents.java

```diff
@@ -47,6 +50,24 @@ public interface FileContents {
      */
     Provider<String> getAsText();
 
+    /**
+     * Gets a provider of the entire file contents as a single String.
+     *
+     * See {@link #getAsText()} for details
+     *
+     * @param charset Charset to be used
+     * @return provider of the entire file contents as a single String.
+     * @since 8.7
+     */
+    @Incubating
+    default Provider<String> getAsText(String charset) {
```

Comment:
Oleg, did we consider taking a `Charset` instead of a string?

Reply:
@abstratt Yes. Using String was @ljacomet's preference. Indeed, I could add the `Charset` implementation too

Reply:
My take was that this is to be exposed in the DSL and when writing tests with Oleg we saw that IDEs will autocomplete charset `String`s.

We can _add_ the `Charset` version, but I slightlyt prefer writing:
`providers.fileContents(layout.projectDirectory.file("file.txt")).getAsText("UTF-8")`
instead of
`providers.fileContents(layout.projectDirectory.file("file.txt")).getAsText(StandardCharsets.UTF_8)`